src/objective.jl

```diff
@@ -99,26 +99,24 @@ function MOI.supports(model::LinQuadOptimizer, ::MOI.ObjectiveFunction{F}) where
     return F in supported_objectives(model)
 end
 
-function MOI.canget(model::LinQuadOptimizer, ::MOI.ObjectiveFunction{MOI.SingleVariable})
-    return model.obj_type == SingleVariableObjective
-end
-
 function MOI.get(model::LinQuadOptimizer, ::MOI.ObjectiveFunction{MOI.SingleVariable})
-    if model.obj_type != MOI.SingleVariable
-        error("Cannot get SingleVariable objective function as the model has " *
-              " a $(model.obj_type) objective function.")
+    if model.obj_type != SingleVariableObjective
+        if VERSION >= v"0.7-"
+            throw(InexactError(:convert, SingleVariableObjective, model.obj_type))            
+        else
+            throw(InexactError())
+        end
```

Comment:
What was wrong with the current error?

Reply:
Just trying to be closer to MOI requirements. https://github.com/JuliaOpt/MathOptInterface.jl/blob/master/src/attributes.jl#L392-L399